app/grandchallenge/algorithms/serializers.py

```diff
@@ -157,6 +154,12 @@ class JobPostSerializer(JobSerializer):
     class Meta:
         model = Job
         fields = ["pk", "algorithm", "inputs", "status"]
+        non_interface_fields = [
+            "algorithm_image",
+            "algorithm_model",
+            "creator",
+            "time_limit",
+        ]
```

Comment:
Ok, I will just give this a try and see how much of the code base this touches if I add this prefix to all interface form fields. It solves a few issues that you commented on in this PR, so it might be worth it if it's not too big a changeset.

Reply:
Done. I'm using the prefix everywhere now and removed `non_interface_fields`.